File: swift/task2/4mod2.swift
import Foundation // Импорт стандартной библиотеки,

func countDigits(_ nums: [Int]) -> Int { // Функция int для счета цифр в числе, принимает массив int как параметр и выводит int
    var count = 0 

    for i in 0..<nums.count { 
        var num = abs(nums[i]) 
        var digits = 0
        if num == 0 {
            digits = 1
        } else {
            while num != 0 {
                num /= 10 
                digits += 1
            }
        }
        if digits % 2 != 0 {
            count += 1
        }
    }

    return count
}

// Main
print("Введите размер массива")

if let input = readLine(), let size = Int(input) {
    var numbers = [Int]() // Пустой массив целых чисел

    for i in 0..<size {
        print("\(i + 1) Элемент массива: ", terminator: "") // Терминатор - не переводим на новую строку
        
        if let elementInput = readLine(), let element = Int(elementInput) {
            numbers.append(element) // Добавляем элемент в массив
        }
    }

    print("\(countDigits(numbers)) Чисел(а) содержат нечетное количество цифр")
}

File: typescript/task2/4mod2.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
const readline = __importStar(require("readline"));
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});
function countDigits(nums) {
    let count = 0;
    for (let i = 0; i < nums.length; i++) {
        let num = Math.abs(nums[i]);
        let digits = 0;
        if (num === 0) {
            digits = 1;
        }
        else {
            while (num !== 0) {
                num = Math.floor(num / 10);
                digits++;
            }
            if (digits % 2 !== 0)
                count++;
        }
    }
    return count;
}
rl.question('Введите размер массива: ', (size) => {
    const sizeNum = parseInt(size, 10);
    const numbers = [];
    let i = 0;
    const next = () => {
        if (i < sizeNum) {
            rl.question(`${i + 1} Элемент массива: `, (element) => {
                numbers.push(parseInt(element, 10));
                i++;
                if (i < sizeNum) { // Запрос до конца элементов
                    next();
                }
                else {
                    const result = countDigits(numbers); // После запроса всех элементов используем функцию countDigits
                    console.log(`${result} Чисел(а) содержат нечетное количество цифр`);
                    rl.close();
                }
            });
        }
    };
    next(); // Запускаем функцию
});
